tools/dataset_converters/yolo2coco.py

```diff
@@ -0,0 +1,162 @@
+import argparse
+import os
+import os.path as osp
+
+import mmcv
+import mmengine
+
+
+def format_by_file(image_dir):
+    """Format annotations by existing train/val/test files."""
+    categories = ['train', 'val', 'test']
+    image_list = []
+
+    for category in categories:
+        txtFile = osp.join(image_dir, f'{category}.txt')
+        print(f'Start to read {category} dataset definition from {txtFile}')
+        assert osp.exists(txtFile)
+
+        with open(txtFile) as f:
+            image_path = f.readlines()
+            image_list.append(image_path)
+    return image_list[0], image_list[1], image_list[2]
+
+
+def convert_yolo_to_coco(image_dir, split=False):
+    """Convert annotations from yolo style to coco style.
+
+    Args:
+        image_dir (str): the root directory of your datasets which contains
+            labels, images, classes.txt, etc
+        split (bool): whether to organize the datasets based on existing
+            train.txt, val.txt, and test.txt
+    """
+    print(f'Start to load existing images and annotations from {image_dir}')
+    assert osp.exists(image_dir)
+
+    yolo_label_dir = osp.join(image_dir, 'labels')
+    yolo_image_dir = osp.join(image_dir, 'images')
+    yolo_class_txt = osp.join(image_dir, 'classes.txt')
+    with open(yolo_class_txt) as f:
+        classes = f.read().strip().split()
+
+    indices = os.listdir(yolo_image_dir)
+
+    if split:
+        print(
+            'Start to organize data according to existing train/test/val split'
+        )
+        train_dataset = {'images': [], 'annotations': [], 'categories': []}
+        val_dataset = {'images': [], 'annotations': [], 'categories': []}
+        test_dataset = {'images': [], 'annotations': [], 'categories': []}
+
+        # categories id starts from 0
+        for i, cls in enumerate(classes, 0):
+            train_dataset['categories'].append({'id': i, 'name': cls})
+            val_dataset['categories'].append({'id': i, 'name': cls})
+            test_dataset['categories'].append({'id': i, 'name': cls})
+        train_img, val_img, test_img = format_by_file(image_dir)
+    else:
+        print('Start to organize data')
+        dataset = {'images': [], 'annotations': [], 'categories': []}
+        for i, cls in enumerate(classes, 0):
+            dataset['categories'].append({'id': i, 'name': cls})
+
+    obj_count = 0
+    for idx, file in enumerate(mmengine.track_iter_progress(indices)):
+        # support both .jpg and .png
+        txtFile = file.replace('images',
```

Comment:
1、`replace` may cause bug, while img filename is `images1.jpg`. Try to use `os.path.splitext`.
2、It is better to use a format list like `['.jpg', '.png', '.jpeg']`  to verify file postfixes.
3、use `.lower()` while verify file postfixes.